api_app/analyzers_manager/file_analyzers/yaraify_file_scan.py

```diff
@@ -35,31 +39,66 @@ def run(self):
             return hash_scan
 
-        data = {
-            "clamav_scan": self.clamav_scan,
-            "unpack": self.unpack,
-            "share_file": self.share_file,
-            "identifier": self.__api_key_identifier,
-        }
-
-        files_ = {
-            "json_data": (None, json.dumps(data), "application/json"),
-            "file": (name_to_send, file),
-        }
-
-        logger.info(f"yara file scan md5 {self.md5} sending sample for analysis")
-        for _try in range(self.max_tries):
-            logger.info(
-                f"yara file scan md5 {self.md5} polling for result try #{_try + 1}"
-            )
+        result = hash_scan
+        if self.send_file:
+            data = {
+                "identifier": self.__api_key_identifier,
+                # the server wants either 0 or 1
+                # https://yaraify.abuse.ch/api/#file-scan
+                "clamav_scan": int(self.clamav_scan),
+                "unpack": int(self.unpack),
+                "share_file": int(self.share_file),
+                "skip_noisy": int(self.skip_noisy),
+                "skip_known": int(self.skip_known),
+            }
+
+            files_ = {
+                "json_data": (None, json.dumps(data), "application/json"),
+                "file": (name_to_send, file),
+            }
+            logger.info(f"yara file scan md5 {self.md5} sending sample for analysis")
             response = requests.post(self.url, files=files_)
-            if response.status_code == 200:
-                break
-            time.sleep(self.poll_distance)
+            response.raise_for_status()
+            scan_response = response.json()
+            scan_query_status = scan_response.get("query_status")
+            if scan_query_status == "queued":
+                task_id = scan_response.get("data", {}).get("task_id", "")
+                if not task_id:
+                    raise AnalyzerRunException(
+                        f"task_id value is unexpected: {task_id}."
+                        f"Analysis was requested for md5 {self.md5}"
+                    )
+                for _try in range(self.max_tries):
+                    try:
+                        logger.info(
+                            f"yara file scan md5 {self.md5} polling for"
+                            f" result try #{_try + 1}."
+                            f"task_id: {task_id}"
+                        )
+                        data = {"query": "get_results", "task_id": task_id}
+                        response = requests.post(self.url, json=data)
+                        response.raise_for_status()
+                        task_response = response.json()
+                        logger.debug(task_response)
+                        data_results = task_response.get("data")
+                        if isinstance(data_results, dict) and data_results:
+                            logger.info(f"found scan result for {self.md5}")
+                            break
```

Comment:
I would have expected that once we have the `data_results`, we always `break` , but if the type is not expected, an exception is raised

Reply:
GG